### PR TITLE
improve compliance refresh token handling

### PR DESCRIPTION
**Pull request by chris-rock**
### Description

This PR is doing the following:
- extract inspec gem installation
- ensures the refresh token is only exchanged once

### Issues Resolved

Customers reported that `refresh_token` handling was not working as expected and lead to authentication errors for some requests.

### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

File: libraries/inspec.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      property :version, String, default: 'latest'
+
+      default_action :install
+
+      # installs inspec if required
+      action :install do
+        converge_by 'install/update inspec' do
+          chef_gem 'inspec' do
+            version version if version != 'latest'
```
Comment (jeremymv2) [inline libraries/inspec.rb]: @chris-rock I get an WARN on line 15 `version` 

Recipe: audit::_inspec
``` 
 * inspec[inspec] action install[2016-08-25T13:28:53+00:00] WARN: /var/chef/cache/cookbooks/audit/libraries/inspec.rb:15:in `block (3 levels) in <class:ChefInspec>': property version is declared in both chef_gem[inspec] and inspec[inspec] action :install. Use new_resource.version instead. At /var/chef/cache/cookbooks/audit/libraries/inspec.rb:15:in `block (3 levels) in <class:ChefInspec>'
[2016-08-25T13:28:53+00:00] WARN: /var/chef/cache/cookbooks/audit/libraries/inspec.rb:15:in `block (3 levels) in <class:ChefInspec>': property version is declared in both chef_gem[inspec] and inspec[inspec] action :install. Use new_resource.version instead. At /var/chef/cache/cookbooks/audit/libraries/inspec.rb:15:in `block (3 levels) in <class:ChefInspec>'
```
This doc seems to support the WARN message above:
https://docs.chef.io/custom_resources.html#new-resource-property

When I test changing line 15 to the following I no longer get the WARNing.
`version new_resource.version if new_resource.version != 'latest'`

Comment (chris-rock) [inline libraries/inspec.rb]: 👍

File: libraries/profile.rb [hunk 2]
```diff
@@ -6,200 +6,178 @@
...
+        converge_by 'fetch compliance profile' do
+          return if path # will be fetched from other source during execute phase
+
+          o, p = normalize_owner_profile
+          Chef::Log.info "Fetch compliance profile #{o}/#{p}"
+
+          path = tar_path
+          access_token = token
+
+          puts "ACCESSTOKEN #{access_token}"
```
Comment (jeremymv2) [inline libraries/profile.rb]: Being a little nit-picky; line 58 looks to be a left-over debugging artifact.  As-is it is appended to the end of current line in console/logs during converges and doesn't have any context and is empty when `['audit']['collector'] == chef-server`.  Perhaps, if this is valuable/desirable info to have in log files (which I'm not totally convinced it is), then a line like: `Chef::Log.info "Using Access Token: #{access_token}" if access_token` would be more delightful for the users?

File: libraries/report.rb [hunk 3]
```diff
@@ -1,108 +1,112 @@
...
+      property :environment, String # default: node.environment
+      property :owner, [String, nil]
+
+      default_action :execute
+
+      action :execute do
+        converge_by "report compliance profiles' results" do
+          reports, ownermap = compound_report(profiles)
+
+          puts "ACCESSTOKEN #{token}"
```
Comment (jeremymv2) [inline libraries/report.rb]: same comment as before about ACCESSTOKEN output.

**Comment by jeremymv2** [comment]
Added a few comments above.  Otherwise this PR looks awesome and will be super helpful.  Tested the PR itself and works like charm!  Very nice @chris-rock!

File: libraries/report.rb (hunk 3 above)
Comment (chris-rock) [inline libraries/report.rb]: 👍 Thanks for highlighting this @jeremymv2

**Pull request merged**